app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,45 @@
+class FoodsController < ApplicationController
+  # require login before accesing the application
+  before_action :authenticate_user!, only: [:index]
+  before_action :require_login, only: [:index]
+
+  # index action for fetching all the food items
+  def index
+    @foods = Food.all
+  end
+
+  # destroy action for deleting a single food item  def destroy
+  def destroy
+    @food = Food.find(params[:id])
+    @food.destroy
+    flash[:notice] = 'Food Item successfully deleted.'
+    redirect_to foods_path
+  end
```

Comment:
- [ ] [OPTIONAL] It would be great if you ensured that we are able to only see the foods that are created by the owner of the foods.
- [ ] [OPTIONAL] Likewise, it is important to ensure that only the owner of the food is able to delete it. At the moment even people who are not the owners of the food are able to delete it.

![image](https://github.com/ismayito/Rails-recipe-app/assets/86318284/d423cd46-df86-469b-80e3-ab8398fa64a2)

Reply:
Fixed.